### PR TITLE
feat: Add support for resource type + tests fix

### PR DESCRIPTION
**Pull request by asafMasa**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

**Review by cl-shaharshavit-rnd — APPROVED** [review]

File: src/commands/generateToken.ts [hunk 1]
```diff
@@ -30,6 +31,7 @@ export const builder: yargs.CommandBuilder<{}, GenerateTokenArguments> = (yargs)
     array: true,
     description: 'the domains that the client will be allowed to make requests from',
   });
+  yargs.option('t', { alias: 'resource-types', type: 'array', array: true, description: 'the product types the client will be allowed to access' });
```
Comment (CptSchnitz) [inline src/commands/generateToken.ts:34]: It should really not be resource/product type as its hard to infer the meaning (rasterized vector is a product type, so is it a valid value?).
something like realm, domain, zone might be better.

File: tests/helpers/execute.ts [hunk 2]
```diff
@@ -16,13 +15,23 @@ const createProcess = (processPath: string, args: string[] = [], env: Env = unde
...
+      stream.on('error', (err) => {
+        reject(err);
+      });
+      stream.on('end', () => {
+        resolve(str);
+      });
+    });
+  } else {
+    return Promise.resolve('');
+  }
```
Comment (CptSchnitz) [inline tests/helpers/execute.ts:34]: it should throw an error. how can you differentiate between empty string as a valid value, and as an error?

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
such a massive changes in tests should be a separate PR. especially when they worked before.
same as RS256 fix. there are 3 different things in the same PR.

File: tests/helpers/execute.ts (hunk 2 above)
Comment (asafMasa) [reply tests/helpers/execute.ts:34]: fixed in tests PR

**Pull request closed without merging**